config/config.properties

```diff
@@ -418,6 +418,29 @@ extendeds3.url=localhost:9020
 
 ##endregion
 
+##region AttributeIndex Settings
+
+# The maximum number of attribute updates processed before a Snapshot is triggered.
+# Valid values: Positive integer.
+# Recommended values: Between 10000 and 100000. Smaller values will cause more frequent snapshots, leading to possibly
+# more churn on the asynchronous StorageWriter but with faster reads, while higher values will cause less frequent snapshots,
+# which could cause increased seek times during reads.
+#attributeindex.updateCountThresholdForSnapshot=50000
```

Comment:
This is on a per-segment basis. Given the aggregation that we do in the StorageWriter, I doubt we'll have more than a few dozen updates per second, per segment.

Consider these extremes:
- Lots of writers, lots of small writes: We'll aggregate and dump to Tier2 when we accumulate 1MB. This will coalesce all those writes/updates into one per writer per 1MB written to the segment.
- Few writers, large writes: Still bounded by the 1MB limit, but the number of attribute updates in TIer2 will be much smaller
- Single writer, many writes: only one update per 1MB written.

The first scenario will case the most updates. If that's the case, then we'll have more frequent snapshots. That's perfectly acceptable, since we want to compress updates into a compact form for easier reading.

And this number is configurable by the customer, if they feel they are not getting a good performance out of the system.